wallet/udb/vsp.go

```diff
@@ -52,6 +94,24 @@ func GetVSPTicket(dbtx walletdb.ReadTx, tickethash chainhash.Hash) (*VSPTicket,
 	}
 	ticket := deserializeVSPTicket(serializedTicket)
 
+	host, err := GetVSPHost(dbtx, ticket.VSPHostID)
+	if err != nil && !errors.Is(err, errors.NotExist) {
+		// Only error out if it's not a 'not exist error'
+		return nil, err
+	}
+	if host != nil && len(host.Host) > 0 {
+		// If the stored host is not empty then get the saved pubkey as well.
+		ticket.Host = string(host.Host)
+		pubkey, err := GetVSPPubKey(dbtx, host.Host)
+		if err != nil && !errors.Is(err, errors.NotExist) {
+			// XXX should we error here when pubkey isn't found but host is?
```

Comment:
I think it's fine as is (i.e. don't error out if there is a host without an associated pubkey) but we might want to add a sanity check earlier in the ticket buying stack to ensure we have both host and valid pubkey before starting a vsp ticket purchase.

Reply:
Good point, I'll add that sanity check.

Reply:
After taking a more in depth look, I believe it's currently impossible to submit a purchase ticket (CLI, grpc with puchase ticket and autobuyer).  All error if vsp host is provided and not pubkey (or vice versa).  So I think we're safe here.